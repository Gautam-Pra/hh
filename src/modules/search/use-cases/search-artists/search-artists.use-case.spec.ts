import { beforeAll, describe, expect, test } from 'vitest'
import { SearchArtistsUseCase } from '#modules/search/use-cases'

describe('SearchArtists', () => {
  let searchArtistsUseCase: SearchArtistsUseCase

  beforeAll(() => {
    searchArtistsUseCase = new SearchArtistsUseCase()
  })

  test('should search artists by query and return a list of artists', async () => {
    const artists = await searchArtistsUseCase.execute({ query: 'imagine dragons', limit: 5, page: 1 })

    expect(artists).toMatchSnapshot()
  })
})
