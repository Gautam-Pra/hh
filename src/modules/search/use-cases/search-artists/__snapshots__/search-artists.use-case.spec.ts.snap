// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchArtists > should search artists by query and return a list of artists 1`] = `
{
  "results": [
    {
      "id": "599452",
      "image": [
        {
          "quality": "50x50",
          "url": "https://c.saavncdn.com/artists/Imagine_Dragons_50x50.jpg",
        },
        {
          "quality": "150x150",
          "url": "https://c.saavncdn.com/artists/Imagine_Dragons_150x150.jpg",
        },
        {
          "quality": "500x500",
          "url": "https://c.saavncdn.com/artists/Imagine_Dragons_500x500.jpg",
        },
      ],
      "name": "Imagine Dragons",
      "role": "Artist",
      "type": "artist",
      "url": "https://www.jiosaavn.com/artist/imagine-dragons-songs/f0aFxsa231o_",
    },
    {
      "id": "4992352",
      "image": [
        {
          "quality": "50x50",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-music.png",
        },
        {
          "quality": "150x150",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-music.png",
        },
        {
          "quality": "500x500",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-music.png",
        },
      ],
      "name": "Kygo &amp; Imagine Dragons",
      "role": "Artist",
      "type": "artist",
      "url": "https://www.jiosaavn.com/artist/kygo-imagine-dragons-songs/KCCw9dHrmMY_",
    },
    {
      "id": "13646546",
      "image": [
        {
          "quality": "50x50",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
        {
          "quality": "150x150",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
        {
          "quality": "500x500",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
      ],
      "name": "INC. OBO IMAGINE DRAGONS PUB",
      "role": "Artist",
      "type": "artist",
      "url": "https://www.jiosaavn.com/artist/inc.-obo-imagine-dragons-pub-songs/oJCAg8iVccM_",
    },
    {
      "id": "13647141",
      "image": [
        {
          "quality": "50x50",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
        {
          "quality": "150x150",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
        {
          "quality": "500x500",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
      ],
      "name": "SONGS OF UNIVERSAL INC OBO IMAGINE DRAGONS PUBLISHING",
      "role": "Artist",
      "type": "artist",
      "url": "https://www.jiosaavn.com/artist/songs-of-universal-inc-obo-imagine-dragons-publishing-songs/B8yvRF,lZ18_",
    },
    {
      "id": "13947978",
      "image": [
        {
          "quality": "50x50",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
        {
          "quality": "150x150",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
        {
          "quality": "500x500",
          "url": "https://www.jiosaavn.com/_i/3.0/artist-default-film.png",
        },
      ],
      "name": "Imagine Dragons Publishing/Songs Of Universal",
      "role": "Artist",
      "type": "artist",
      "url": "https://www.jiosaavn.com/artist/imagine-dragons-publishingsongs-of-universal-songs/29zV6ilDdl0_",
    },
  ],
  "start": 1,
  "total": 13,
}
`;
